Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1274 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// Originally written by Sven Peter <sven@fail0verflow.com> for anergistic.
+
+#include "Core/PowerPC/GDBThread.h"
+
+#include "Common/CommonTypes.h"
+#include "Common/Event.h"
+#include "Common/Logging/Log.h"
+#include "Common/Thread.h"
+#include "Core/BootManager.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "Core/HW/CPU.h"
+#include "Core/HW/Memmap.h"
+#include "Core/Host.h"
+#include "Core/PowerPC/Interpreter/Interpreter_FPUtils.h"
+#include "Core/PowerPC/JitInterface.h"
+#include "Core/PowerPC/PowerPC.h"
+
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <string.h>
+#ifdef _WIN32
+#include <iphlpapi.h>
+#include <ws2tcpip.h>
+#else
+#include <netinet/in.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+#define SD_BOTH SHUT_RDWR
+#endif
+#include <stdarg.h>
+
+#define fail(msg)                                                                                  \
+  {                                                                                                \
+    ERROR_LOG(GDB_THREAD, msg);                                                                    \
+    GDBDeinit();                                                                                   \
+    return;                                                                                        \
+  }
+
+#define failr(msg)                                                                                 \
```

Comment:
Any thoughts on making those into methods? Using defines for this purpose is kinda last century :D

Reply:
Hold over from plugin based on older C based code sample where implementation was first made and this was merged from.  Believe i fixed this in all other versions of this except for the one for Dolphin, heh.